testdata/jennies/builders/struct_with_defaults/schema.cue

```diff
@@ -0,0 +1,17 @@
+package struct_with_defaults
+
+#NestedStruct: {
+	stringVal: string
+	intVal: int64
+}
+
+#Struct: {
+	allFields: #NestedStruct | *{
+  	stringVal: "hello"
+  	intVal: 3
+  }
+  partialFields: #NestedStruct | *{
+  	intVal: 4
+  }
+  emptyFields: #NestedStruct | *{}
+}
```

Comment:
Should we add a test case with anonymous structs?

```cue
#SomeStruct: {
	complexField: {
		intVal: int64
	} | *{ intVal: 2 }
}
```

Reply:
Added :)